pkg/controller/testutil/test_utils.go

```diff
@@ -480,3 +487,27 @@ func GetZones(nodeHandler *FakeNodeHandler) []string {
 func CreateZoneID(region, zone string) string {
 	return region + ":\x00:" + zone
 }
+
+// GetKey is a helper function used by controllers unit tests to get the
+// key for a given kubernetes resource.
+func GetKey(obj interface{}, t *testing.T) string {
+	tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+	if ok {
+		// if tombstone , try getting the value from tombstone.Obj
+		obj = tombstone.Obj
+	}
+	val := reflect.ValueOf(obj).Elem()
+	name := val.FieldByName("Name").String()
+	kind := val.FieldByName("Kind").String()
+	// Note kind is not always set in the tests, so ignoring that for now
```

Comment:
You can add TypeMeta.Kind to resources used in tests to avoid this "kind not set" error.

Reply:
ok let me check, i think there will be too many tests to fix